vnext/Microsoft.ReactNative/Views/ViewManagerBase.h

```diff
@@ -101,6 +105,7 @@ class REACTWINDOWS_EXPORT ViewManagerBase : public IViewManager {
 
  protected:
   Mso::CntPtr<const Mso::React::IReactContext> m_context;
+  std::shared_ptr<winrt::Microsoft::ReactNative::BatchingEventEmitter> m_batchingEventEmitter;
```

Comment:
Adding the comment from previous discussions for reference here:

This places an instance of the `batchingEmitter` in each VM - does it make sense to move this to the context or someplace else?

Reply:
I have mixed feelings. There is less memory usage, for less emitters, but the queue length for a batch goes up, which makes coalescing potentially more expansive.

Reply:
More efficient data structure is definitely possible though.

Reply:
The logic we copied from ReactAndroid in the C# bridge used to keep a map from the "event cookie" to the array index of the event to coalesce. The event cookie was an integer created from the event name and coalescing key. And rather than iterating over the queued events, it just did a constant time lookup of the current index of the last matching event to coalesce against.

Reply:
It's a good point though, especially for cases of many views calling onLayout in the same batch, the queue will get quite long. We shouldn't push this until we come up with a more efficient coalesce matching mechanism.

Reply:
My guess is a per-ViewManager queue should not have more than a hundred queued events per-batch. I don't have any qualms using a simpler structure over an optimized one if we bucket the overall system (like per-vm).

Reply:
I added a simple map from `(eventId, coalescingKey) -> lastEventIndex` for constant time lookup. This should allow us to share the batching emitter across all view managers.